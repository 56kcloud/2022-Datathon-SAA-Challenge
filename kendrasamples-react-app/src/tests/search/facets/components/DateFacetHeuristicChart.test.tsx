import { shallow } from "enzyme";
import React from "react";
import { DateFacetHeuristicChart } from "../../../../search/facets/components/DateFacetHeuristicChart";

describe("A <DateFacetHeuristicChart />", () => {
  it("Renders correctly", () => {
    const component = shallow(
      <DateFacetHeuristicChart
        availableDateFacet={{
          minYear: 2018,
          maxYear: 2020,
          yearHeuristic: {
            2018: 5,
            2019: 0,
            2020: 10,
          },
        }}
        displaySelectionRange={[2019, 2020]}
        fullRange={[2018, 2020]}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
