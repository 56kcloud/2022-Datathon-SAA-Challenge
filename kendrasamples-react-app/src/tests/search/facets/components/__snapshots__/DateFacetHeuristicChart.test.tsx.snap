// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <DateFacetHeuristicChart /> Renders correctly 1`] = `
<Chart
  chartActions={null}
  chartEvents={null}
  chartType="ColumnChart"
  chartWrapperParams={Object {}}
  className=""
  columns={null}
  controls={null}
  data={
    Array [
      Array [
        Object {
          "type": "string",
        },
        "Values",
        Object {
          "role": "style",
        },
      ],
      Array [
        "2018",
        5,
        "color: #d5dbdb",
      ],
      Array [
        "2019",
        0,
        "color: #007bff",
      ],
      Array [
        "2020",
        10,
        "color: #007bff",
      ],
    ]
  }
  diffdata={null}
  formatters={null}
  getChartEditor={null}
  getChartWrapper={[Function]}
  graphID={null}
  graph_id={null}
  height="75px"
  legendToggle={false}
  legend_toggle={false}
  options={
    Object {
      "bar": Object {
        "groupWidth": "90%",
      },
      "chartArea": Object {
        "width": "100%",
      },
      "focusTarget": "category",
      "hAxis": Object {
        "baselineColor": "transparent",
        "gridlines": Object {
          "color": "transparent",
        },
        "textPosition": "none",
      },
      "legend": Object {
        "position": "none",
      },
      "tooltip": Object {
        "isHtml": true,
        "showColorCode": false,
      },
      "vAxis": Object {
        "baselineColor": "transparent",
        "gridlines": Object {
          "color": "transparent",
        },
        "textPosition": "none",
      },
    }
  }
  render={null}
  rootProps={Object {}}
  rows={null}
  spreadSheetQueryParameters={
    Object {
      "gid": 1,
      "headers": 1,
    }
  }
  spreadSheetUrl={null}
  style={Object {}}
  toolbarID={null}
  toolbarItems={null}
  width="100%"
/>
`;
